### ISSUE TITLE
Assets Subsystem needs to honor order instead of weight.

### ISSUE THREAD

**Issue by aembler** [concrete5/concrete5-legacy#1467]
Currently, the new concrete5 assets system has a weight parameter which is used primarily to show that jQuery is important and needs to come before anything else. We don't really need this, nor do we really need to add in a "requires" concept. We already HAVE a requires concept – it's the grouping of assets in the AssetGroup. What we nee to do is ensure that the assets come in the groups in the proper order, and that that order is translated into a weight parameter behind the scenes. Then, you just make sure jquery.js comes before any other file in your various asset groups and voila, it just works.

To recap:
1. Remove "weight" as a parameter in config/assets.php and in the assets models.
2. Honor the order of items in the assetgroups.
3. Make sure that jQuery is first in all its asset groups.
4. Ensure that concrete5 still works.
